austation/code/game/gamemodes/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,296 @@
+	// The antags you're allowed to be if turning Vassal.
+
+/datum/game_mode/bloodsucker
+	name = "bloodsucker"
+	config_tag = "bloodsucker"
+	report_type = "traitor"
+	traitor_name = "Bloodsucker"//Nanotrasen Internal Affairs Agent"
+	antag_flag = ROLE_BLOODSUCKER
+	false_report_weight = 1
+	restricted_jobs = list("AI","Cyborg")
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Deputy")
+	required_players = 0
+	required_enemies = 1
+	recommended_enemies = 4
+	reroll_friendly = 1
+	enemy_minimum_age = 7
+
+	announce_span = "danger"
+	announce_text = "Filthy, bloodsucking vampires are crawling around disguised as crewmembers!\n\
+	<span class='danger'>Bloodsuckers</span>: The crew are cattle, while you are both shepherd and slaughterhouse.\n\
+	<span class='notice'>Crew</span>: Put an end to the undead infestation before the station is overcome!"
+
+
+
+// Seems to be run by game ONCE, and finds all potential players to be antag.
+/datum/game_mode/bloodsucker/pre_setup()
+
+	// Set Restricted Jobs
+	if(CONFIG_GET(flag/protect_roles_from_antagonist))
+		restricted_jobs += protected_jobs
+
+	if(CONFIG_GET(flag/protect_assistant_from_antagonist))
+		restricted_jobs += "Assistant"
+
+	// Set number of Vamps
+	recommended_enemies = clamp(round(num_players()/10), 1, 6);
+
+	// Select Antags
+	for(var/i = 0, i < recommended_enemies, i++)
+		if (!antag_candidates.len)
+			break
+		var/datum/mind/bloodsucker = pick(antag_candidates)
+		// Can we even BE a bloodsucker?
+		//if (can_make_bloodsucker(bloodsucker, display_warning=FALSE))
+		bloodsuckers += bloodsucker
+		bloodsucker.restricted_roles = restricted_jobs
+		log_game("[bloodsucker.key] (ckey) has been selected as a Bloodsucker.")
+		antag_candidates.Remove(bloodsucker) // Apparently you can also write antag_candidates -= bloodsucker
+
+	// FULPSTATION: Assign Hunters (as many as monsters, plus one)
+	//assign_monster_hunters(bloodsuckers.len, TRUE, bloodsuckers)	// austation -- DISABLING HUNTERS FROM ROUND START, We have enough validhunters anyway
+
+	// Do we have enough vamps to continue?
+	return bloodsuckers.len >= required_enemies
+
+
+// Gamemode is all done being set up. We have all our Vamps. We now pick objectives and let them know what's happening.
+/datum/game_mode/bloodsucker/post_setup()
+
+	// Sunlight (Creating Bloodsuckers manually will check to create this, too)
+	check_start_sunlight()
+
+	// Vamps
+	for(var/datum/mind/bloodsucker in bloodsuckers)
+		// spawn() --> Run block of code but game continues on past it.
+		// sleep() --> Run block of code and freeze code there (including whoever called us) until it's resolved.
+
+		//Clean Bloodsucker Species (racist?)
+		//clean_invalid_species(bloodsucker)
+		// 			TO-DO !!!
+
+		// Add Bloodsucker Antag Datum (or remove from list on Fail)
+		if (!make_bloodsucker(bloodsucker))
+			bloodsuckers -= bloodsucker
+
+	// NOTE: Hunters are done in ..() parent proc
+
```

Comment:
```suggestion

	if (!bloodsuckers.len)
		message_admins("Failed to make bloodsuckers, despite bloodsucker game_mode. Please rectify this situation.")

```
This literally is a band-aid in case of lowpop.

Reply:
I think this is in all gamemodes. If it can't create the antagonists it messages the admins. Besides all it requires is one person with a correct race, if the server is playing with at least 10 people there will be at least 1 felinid roaming around ready to give people the SUCC